internal/schema/templates/nested_object_value_to_object_value.gotmpl

```diff
@@ -111,7 +110,18 @@ AttrTypes: v.{{$key.ToPrefixPascalCase $.Name}}.AttributeTypes(ctx),
 "{{$key}}": {{index $.AttrTypes $key}},
 {{- end}}
 {{- end}}
-},
+}
+
+if v.state == attr.ValueStateNull {
```

Comment:
It's a little awkward since this function isn't defined in this `gotmpl` file, but we can essentially guarantee that `v` will have `v.IsNull() bool` and `v.IsUnknown() bool` functions. All custom type values implement `attr.Value` which require these implementations: https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework@v1.8.0/attr#Value

So let's use `v.IsNull()` here and `v.IsUnknown()` below

Reply:
Ah, makes sense! Pushed a commit with this change 👍